clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -300,18 +312,24 @@ public void run() {
                         } finally {
                             hubConnectionStateLock.unlock();
                         }
-                    });
-                });
-            });
-        }));
+
+                        return Completable.complete();
+                    }));
+                }));
+            }));
+        // subscribe makes this a "hot" completable so this runs immediately
+        }).subscribe(() -> start.onComplete(),
```

Comment:
I think you can just use `.subscribeWith(start)` since `CompletableSubject` implements `CompletableObserver`

Reply:
Oh neat!